### PR TITLE
`dns`: Added query type.

### PR DESCRIPTION
**Pull request by eiffel-fl**
Hi.

In this PR, I added query type to `dns` output:

```bash
$ ./kubectl-gadget dns -n demo &
POD                            TYPE      QTYPE     NAME
$ kubectl -n demo run mypod -it --image=praqma/network-multitool -- /bin/sh
# nslookup microsoft.com > /dev/null
mypod                          OUTGOING  A         microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  A         microsoft.com.svc.cluster.local.
mypod                          OUTGOING  A         microsoft.com.cluster.local.
mypod                          OUTGOING  AAAA  microsoft.com.
```

<details>
  <summary>To test all DNS request type, run the following:</summary>
  
  ```bash
$ ./kubectl-gadget dns -n demo &
POD                            TYPE      QTYPE     NAME
$ kubectl -n demo run mypod -it --image=praqma/network-multitool -- /bin/sh
# for request in a ns md mf cname soa mb mg mr null wks ptr hinfo minfo mx txt rp afsdb x25 isdn rt nsap nsap-ptr sig key px gpos aaaa loc nxt 
eid nimloc srv atma naptr kx cert a6 dname sink opt apl ds sshfp ipseckey rrsig nsec dnskey dhcid nsec3 nsec3param tlsa smimea hip ninfo rkey tal
ink cds cdnskey openpgpkey csync zonemd svcb https spf uinfo uid gid unspec nid l32 l64 lp eui48 eui64 tkey tsig ixfr axfr mailb maila uri caa av
c doa amtrelay ta dlv; do
> nslookup -type=$request microsoft.com > /dev/null
> done
mypod                          OUTGOING  A         microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  A         microsoft.com.svc.cluster.local.
mypod                          OUTGOING  A         microsoft.com.cluster.local.
mypod                          OUTGOING  A         microsoft.com.
mypod                          OUTGOING  NS        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  NS        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  NS        microsoft.com.cluster.local.
mypod                          OUTGOING  NS        microsoft.com.
mypod                          OUTGOING  MD        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  MD        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  MD        microsoft.com.cluster.local.
mypod                          OUTGOING  MD        microsoft.com.
mypod                          OUTGOING  MF        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  MF        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  MF        microsoft.com.cluster.local.
mypod                          OUTGOING  MF        microsoft.com.
mypod                          OUTGOING  CNAME     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  CNAME     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  CNAME     microsoft.com.cluster.local.
mypod                          OUTGOING  CNAME     microsoft.com.
mypod                          OUTGOING  SOA       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  MB        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  MB        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  MB        microsoft.com.cluster.local.
mypod                          OUTGOING  MB        microsoft.com.
mypod                          OUTGOING  MG        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  MG        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  MG        microsoft.com.cluster.local.
mypod                          OUTGOING  MG        microsoft.com.
mypod                          OUTGOING  MR        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  MR        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  MR        microsoft.com.cluster.local.
mypod                          OUTGOING  MR        microsoft.com.
mypod                          OUTGOING  NULL      microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  NULL      microsoft.com.svc.cluster.local.
mypod                          OUTGOING  NULL      microsoft.com.cluster.local.
mypod                          OUTGOING  NULL      microsoft.com.
mypod                          OUTGOING  WKS       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  WKS       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  WKS       microsoft.com.cluster.local.
mypod                          OUTGOING  WKS       microsoft.com.
mypod                          OUTGOING  PTR       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  PTR       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  PTR       microsoft.com.cluster.local.
mypod                          OUTGOING  PTR       microsoft.com.
mypod                          OUTGOING  HINFO     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  HINFO     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  HINFO     microsoft.com.cluster.local.
mypod                          OUTGOING  HINFO     microsoft.com.
mypod                          OUTGOING  MINFO     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  MINFO     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  MINFO     microsoft.com.cluster.local.
mypod                          OUTGOING  MINFO     microsoft.com.
mypod                          OUTGOING  MX        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  MX        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  MX        microsoft.com.cluster.local.
mypod                          OUTGOING  MX        microsoft.com.
mypod                          OUTGOING  TXT       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  RP        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  RP        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  RP        microsoft.com.cluster.local.
mypod                          OUTGOING  RP        microsoft.com.
mypod                          OUTGOING  AFSDB     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  AFSDB     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  AFSDB     microsoft.com.cluster.local.
mypod                          OUTGOING  AFSDB     microsoft.com.
mypod                          OUTGOING  X25       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  X25       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  X25       microsoft.com.cluster.local.
mypod                          OUTGOING  X25       microsoft.com.
mypod                          OUTGOING  ISDN      microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  ISDN      microsoft.com.svc.cluster.local.
mypod                          OUTGOING  ISDN      microsoft.com.cluster.local.
mypod                          OUTGOING  ISDN      microsoft.com.
mypod                          OUTGOING  RT        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  RT        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  RT        microsoft.com.cluster.local.
mypod                          OUTGOING  RT        microsoft.com.
mypod                          OUTGOING  NSAP      microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  NSAP      microsoft.com.svc.cluster.local.
mypod                          OUTGOING  NSAP      microsoft.com.cluster.local.
mypod                          OUTGOING  NSAP      microsoft.com.
mypod                          OUTGOING  NSAP-PTR  microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  NSAP-PTR  microsoft.com.svc.cluster.local.
mypod                          OUTGOING  NSAP-PTR  microsoft.com.cluster.local.
mypod                          OUTGOING  NSAP-PTR  microsoft.com.
mypod                          OUTGOING  SIG       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  SIG       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  SIG       microsoft.com.cluster.local.
mypod                          OUTGOING  SIG       microsoft.com.
mypod                          OUTGOING  KEY       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  KEY       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  KEY       microsoft.com.cluster.local.
mypod                          OUTGOING  KEY       microsoft.com.
mypod                          OUTGOING  PX        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  PX        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  PX        microsoft.com.cluster.local.
mypod                          OUTGOING  PX        microsoft.com.
mypod                          OUTGOING  GPOS      microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  GPOS      microsoft.com.svc.cluster.local.
mypod                          OUTGOING  GPOS      microsoft.com.cluster.local.
mypod                          OUTGOING  GPOS      microsoft.com.
mypod                          OUTGOING  AAAA      microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  AAAA      microsoft.com.svc.cluster.local.
mypod                          OUTGOING  AAAA      microsoft.com.cluster.local.
mypod                          OUTGOING  AAAA      microsoft.com.
mypod                          OUTGOING  LOC       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  LOC       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  LOC       microsoft.com.cluster.local.
mypod                          OUTGOING  LOC       microsoft.com.
mypod                          OUTGOING  NXT       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  NXT       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  NXT       microsoft.com.cluster.local.
mypod                          OUTGOING  NXT       microsoft.com.
mypod                          OUTGOING  EID       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  EID       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  EID       microsoft.com.cluster.local.
mypod                          OUTGOING  EID       microsoft.com.
mypod                          OUTGOING  NIMLOC    microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  NIMLOC    microsoft.com.svc.cluster.local.
mypod                          OUTGOING  NIMLOC    microsoft.com.cluster.local.
mypod                          OUTGOING  NIMLOC    microsoft.com.
mypod                          OUTGOING  SRV       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  SRV       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  SRV       microsoft.com.cluster.local.
mypod                          OUTGOING  SRV       microsoft.com.
mypod                          OUTGOING  ATMA      microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  ATMA      microsoft.com.svc.cluster.local.
mypod                          OUTGOING  ATMA      microsoft.com.cluster.local.
mypod                          OUTGOING  ATMA      microsoft.com.
mypod                          OUTGOING  NAPTR     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  NAPTR     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  NAPTR     microsoft.com.cluster.local.
mypod                          OUTGOING  NAPTR     microsoft.com.
mypod                          OUTGOING  KX        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  KX        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  KX        microsoft.com.cluster.local.
mypod                          OUTGOING  KX        microsoft.com.
mypod                          OUTGOING  CERT      microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  CERT      microsoft.com.svc.cluster.local.
mypod                          OUTGOING  CERT      microsoft.com.cluster.local.
mypod                          OUTGOING  CERT      microsoft.com.
mypod                          OUTGOING  A6        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  A6        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  A6        microsoft.com.cluster.local.
mypod                          OUTGOING  A6        microsoft.com.
mypod                          OUTGOING  DNAME     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  DNAME     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  DNAME     microsoft.com.cluster.local.
mypod                          OUTGOING  DNAME     microsoft.com.
mypod                          OUTGOING  SINK      microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  SINK      microsoft.com.svc.cluster.local.
mypod                          OUTGOING  SINK      microsoft.com.cluster.local.
mypod                          OUTGOING  SINK      microsoft.com.
mypod                          OUTGOING  OPT       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  OPT       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  OPT       microsoft.com.cluster.local.
mypod                          OUTGOING  OPT       microsoft.com.
mypod                          OUTGOING  APL       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  APL       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  APL       microsoft.com.cluster.local.
mypod                          OUTGOING  APL       microsoft.com.
mypod                          OUTGOING  DS        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  DS        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  DS        microsoft.com.cluster.local.
mypod                          OUTGOING  DS        microsoft.com.
mypod                          OUTGOING  SSHFP     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  SSHFP     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  SSHFP     microsoft.com.cluster.local.
mypod                          OUTGOING  SSHFP     microsoft.com.
mypod                          OUTGOING  IPSECKEY  microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  IPSECKEY  microsoft.com.svc.cluster.local.
mypod                          OUTGOING  IPSECKEY  microsoft.com.cluster.local.
mypod                          OUTGOING  IPSECKEY  microsoft.com.
mypod                          OUTGOING  RRSIG     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  RRSIG     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  RRSIG     microsoft.com.cluster.local.
mypod                          OUTGOING  RRSIG     microsoft.com.
mypod                          OUTGOING  NSEC      microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  NSEC      microsoft.com.svc.cluster.local.
mypod                          OUTGOING  NSEC      microsoft.com.cluster.local.
mypod                          OUTGOING  NSEC      microsoft.com.
mypod                          OUTGOING  DNSKEY    microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  DNSKEY    microsoft.com.svc.cluster.local.
mypod                          OUTGOING  DNSKEY    microsoft.com.cluster.local.
mypod                          OUTGOING  DNSKEY    microsoft.com.
mypod                          OUTGOING  DHCID     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  DHCID     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  DHCID     microsoft.com.cluster.local.
mypod                          OUTGOING  DHCID     microsoft.com.
mypod                          OUTGOING  NSEC3     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  NSEC3     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  NSEC3     microsoft.com.cluster.local.
mypod                          OUTGOING  NSEC3     microsoft.com.
mypod                          OUTGOING  NSEC3PARA microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  NSEC3PARA microsoft.com.svc.cluster.local.
mypod                          OUTGOING  NSEC3PARA microsoft.com.cluster.local.
mypod                          OUTGOING  NSEC3PARA microsoft.com.
mypod                          OUTGOING  TLSA      microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  TLSA      microsoft.com.svc.cluster.local.
mypod                          OUTGOING  TLSA      microsoft.com.cluster.local.
mypod                          OUTGOING  TLSA      microsoft.com.
mypod                          OUTGOING  SMIMEA    microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  SMIMEA    microsoft.com.svc.cluster.local.
mypod                          OUTGOING  SMIMEA    microsoft.com.cluster.local.
mypod                          OUTGOING  SMIMEA    microsoft.com.
mypod                          OUTGOING  HIP       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  HIP       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  HIP       microsoft.com.cluster.local.
mypod                          OUTGOING  HIP       microsoft.com.
mypod                          OUTGOING  NINFO     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  NINFO     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  NINFO     microsoft.com.cluster.local.
mypod                          OUTGOING  NINFO     microsoft.com.
mypod                          OUTGOING  RKEY      microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  RKEY      microsoft.com.svc.cluster.local.
mypod                          OUTGOING  RKEY      microsoft.com.cluster.local.
mypod                          OUTGOING  RKEY      microsoft.com.
mypod                          OUTGOING  TALINK    microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  TALINK    microsoft.com.svc.cluster.local.
mypod                          OUTGOING  TALINK    microsoft.com.cluster.local.
mypod                          OUTGOING  TALINK    microsoft.com.
mypod                          OUTGOING  CDS       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  CDS       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  CDS       microsoft.com.cluster.local.
mypod                          OUTGOING  CDS       microsoft.com.
mypod                          OUTGOING  CDNSKEY   microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  CDNSKEY   microsoft.com.svc.cluster.local.
mypod                          OUTGOING  CDNSKEY   microsoft.com.cluster.local.
mypod                          OUTGOING  CDNSKEY   microsoft.com.
mypod                          OUTGOING  OPENPGPKE microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  OPENPGPKE microsoft.com.svc.cluster.local.
mypod                          OUTGOING  OPENPGPKE microsoft.com.cluster.local.
mypod                          OUTGOING  OPENPGPKE microsoft.com.
mypod                          OUTGOING  CSYNC     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  CSYNC     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  CSYNC     microsoft.com.cluster.local.
mypod                          OUTGOING  CSYNC     microsoft.com.
mypod                          OUTGOING  ZONEMD    microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  ZONEMD    microsoft.com.svc.cluster.local.
mypod                          OUTGOING  ZONEMD    microsoft.com.cluster.local.
mypod                          OUTGOING  ZONEMD    microsoft.com.
mypod                          OUTGOING  A         microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  A         microsoft.com.svc.cluster.local.
mypod                          OUTGOING  A         microsoft.com.cluster.local.
mypod                          OUTGOING  A         microsoft.com.
mypod                          OUTGOING  AAAA      microsoft.com.
mypod                          OUTGOING  A         microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  A         microsoft.com.svc.cluster.local.
mypod                          OUTGOING  A         microsoft.com.cluster.local.
mypod                          OUTGOING  A         microsoft.com.
mypod                          OUTGOING  AAAA      microsoft.com.
mypod                          OUTGOING  SPF       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  SPF       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  SPF       microsoft.com.cluster.local.
mypod                          OUTGOING  SPF       microsoft.com.
mypod                          OUTGOING  UINFO     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  UINFO     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  UINFO     microsoft.com.cluster.local.
mypod                          OUTGOING  UINFO     microsoft.com.
mypod                          OUTGOING  UID       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  UID       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  UID       microsoft.com.cluster.local.
mypod                          OUTGOING  UID       microsoft.com.
mypod                          OUTGOING  GID       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  GID       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  GID       microsoft.com.cluster.local.
mypod                          OUTGOING  GID       microsoft.com.
mypod                          OUTGOING  UNSPEC    microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  UNSPEC    microsoft.com.svc.cluster.local.
mypod                          OUTGOING  UNSPEC    microsoft.com.cluster.local.
mypod                          OUTGOING  UNSPEC    microsoft.com.
mypod                          OUTGOING  NID       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  NID       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  NID       microsoft.com.cluster.local.
mypod                          OUTGOING  NID       microsoft.com.
mypod                          OUTGOING  L32       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  L32       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  L32       microsoft.com.cluster.local.
mypod                          OUTGOING  L32       microsoft.com.
mypod                          OUTGOING  L64       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  L64       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  L64       microsoft.com.cluster.local.
mypod                          OUTGOING  L64       microsoft.com.
mypod                          OUTGOING  LP        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  LP        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  LP        microsoft.com.cluster.local.
mypod                          OUTGOING  LP        microsoft.com.
mypod                          OUTGOING  EUI48     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  EUI48     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  EUI48     microsoft.com.cluster.local.
mypod                          OUTGOING  EUI48     microsoft.com.
mypod                          OUTGOING  EUI64     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  EUI64     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  EUI64     microsoft.com.cluster.local.
mypod                          OUTGOING  EUI64     microsoft.com.
mypod                          OUTGOING  TKEY      microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  TKEY      microsoft.com.svc.cluster.local.
mypod                          OUTGOING  TKEY      microsoft.com.cluster.local.
mypod                          OUTGOING  TKEY      microsoft.com.
mypod                          OUTGOING  TSIG      microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  TSIG      microsoft.com.svc.cluster.local.
mypod                          OUTGOING  TSIG      microsoft.com.cluster.local.
mypod                          OUTGOING  TSIG      microsoft.com.
mypod                          HOST      IXFR      microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  MAILB     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  MAILB     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  MAILB     microsoft.com.cluster.local.
mypod                          OUTGOING  MAILB     microsoft.com.
mypod                          OUTGOING  MAILA     microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  MAILA     microsoft.com.svc.cluster.local.
mypod                          OUTGOING  MAILA     microsoft.com.cluster.local.
mypod                          OUTGOING  MAILA     microsoft.com.
mypod                          OUTGOING  URI       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  URI       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  URI       microsoft.com.cluster.local.
mypod                          OUTGOING  URI       microsoft.com.
mypod                          OUTGOING  CAA       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  CAA       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  CAA       microsoft.com.cluster.local.
mypod                          OUTGOING  CAA       microsoft.com.
mypod                          OUTGOING  AVC       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  AVC       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  AVC       microsoft.com.cluster.local.
mypod                          OUTGOING  AVC       microsoft.com.
mypod                          OUTGOING  DOA       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  DOA       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  DOA       microsoft.com.cluster.local.
mypod                          OUTGOING  DOA       microsoft.com.
mypod                          OUTGOING  AMTRELAY  microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  AMTRELAY  microsoft.com.svc.cluster.local.
mypod                          OUTGOING  AMTRELAY  microsoft.com.cluster.local.
mypod                          OUTGOING  AMTRELAY  microsoft.com.
mypod                          OUTGOING  TA        microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  TA        microsoft.com.svc.cluster.local.
mypod                          OUTGOING  TA        microsoft.com.cluster.local.
mypod                          OUTGOING  TA        microsoft.com.
mypod                          OUTGOING  DLV       microsoft.com.demo.svc.cluster.local.
mypod                          OUTGOING  DLV       microsoft.com.svc.cluster.local.
mypod                          OUTGOING  DLV       microsoft.com.cluster.local.
mypod                          OUTGOING  DLV       microsoft.com.
  ```
  
</details>

Best regards.

### CODE REVIEW

File: pkg/gadgets/dns/tracer/tracer.go [hunk 1]
```diff
@@ -194,10 +288,19 @@ func parseDNSEvent(rawSample []byte) (ret string, pktType string) {
...
 	}
 
+	qTypeUint := uint(dnsEvent.qtype)
+	// Magical values comes from:
+	// https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-4
+	if qTypeUint == 54 || (66 <= qTypeUint && qTypeUint <= 98) || (110 <= qTypeUint && qTypeUint <= 248) || (261 <= qTypeUint && qTypeUint <= 32767) || 32770 <= qTypeUint {
+		qType = "UNASSIGNED"
+	} else {
+		qType = qTypeNames[qTypeUint]
+	}
```
Comment (mauriciovasquezbernal) [inline pkg/gadgets/dns/tracer/tracer.go:298]: What about doing something like:

```
qType, ok := qTypeNames[qTypeUint]
if !ok {
    qType = "UNASSIGNED"
}
```

File: pkg/gadgets/dns/tracer/tracer.go [hunk 2]
```diff
@@ -167,7 +167,101 @@ var pktTypeNames = []string{
 	"KERNEL",
 }
 
-func parseDNSEvent(rawSample []byte) (ret string, pktType string) {
+// List taken from:
+// https://www.iana.org/assignments/dns-parameters/dns-parameters.xhtml#dns-parameters-4
+var qTypeNames = []string{
```
Comment (mauriciovasquezbernal) [inline pkg/gadgets/dns/tracer/tracer.go:172]:
```suggestion
var qTypeNames = map[int]string{
```

I'm not sure what the difference is but I think the second one is more readable.

**Review by mauriciovasquezbernal — COMMENTED** [review]
LGTM. Two minor comments.

File: pkg/gadgets/dns/tracer/tracer.go [hunk 3]
```diff
@@ -167,7 +167,101 @@ var pktTypeNames = []string{
...
+	1:     "A",
```
Comment (blanquicet) [inline pkg/gadgets/dns/tracer/tracer.go:173]: I am just wondering if we should use `A-IPv4` and `AAAA-IPv6` or maybe just `IPv4` and `IPv6`. Not everybody knows  `A` and `AAAA` means that.

**Review by blanquicet — COMMENTED** [review]
I have just one concern about IPv4 and IPv6 query type names. Otherwise LGTM.

Fixes #254

File: pkg/gadgets/dns/tracer/tracer.go (hunk 1 above)
Comment (eiffel-fl) [reply pkg/gadgets/dns/tracer/tracer.go:298]: Thank you for it, it is far cleaner than what I wrote (I was too `C`-focused).

File: pkg/gadgets/dns/tracer/tracer.go (hunk 2 above)
Comment (eiffel-fl) [reply pkg/gadgets/dns/tracer/tracer.go:172]: More readable and I also think it takes less spaces, thank you!

File: pkg/gadgets/dns/tracer/tracer.go (hunk 3 above)
Comment (mauriciovasquezbernal) [reply pkg/gadgets/dns/tracer/tracer.go]: I think people are familiar enough with `A` and `AAAA`. I won't deviate from the standard representation of such queries.

Comment (eiffel-fl) [reply pkg/gadgets/dns/tracer/tracer.go]: As a middle ground, the last version of the patch shows `A (IPv4)` and `AAAA (IPv6)`.

Comment (mauriciovasquezbernal) [reply pkg/gadgets/dns/tracer/tracer.go]: I still think `A` and `AAAA` are better, it's the standard and I don't see any reason why to use something different. Additionally IMO `A` and `AAAA` are widely known and we should not worry about Jose's concern above.

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM. I made a comment regarding query name: https://github.com/kinvolk/inspektor-gadget/pull/461#discussion_r784820075 but it's not blocking. If you think the current way is better I'm ok with that.

File: pkg/gadgets/dns/tracer/tracer.go (hunk 3 above)
Comment (eiffel-fl) [reply pkg/gadgets/dns/tracer/tracer.go:172]: I finally went with the standard.
It is possible people will not know it but it is not so complicated to learn superficially to understand `A` and `AAAA`.

**Pull request merged**